stix2/sources/__init__.py

```diff
@@ -136,194 +221,122 @@ def add(self, stix_objs):
 
 
 class DataSource(object):
-    """
-    Abstract class for defining a data source. Intended for subclassing into
-    different source components.
+    """DataSource
+
+    An implementer will create a concrete subclass from
+    this class for the specific DataSource.
 
     Attributes:
         id (str): A unique UUIDv4 to identify this DataSource.
-        filters (set): A collection of filters present in this DataSource.
+
+        _filters (set): A collection of filters attached to this DataSource.
 
     """
     def __init__(self):
         self.id = make_id()
-        self.filters = set()
+        self._filters = set()
 
     def get(self, stix_id, _composite_filters=None):
         """
-        Fill:
-            Implement the specific data source API calls, processing,
-            functionality required for retrieving data from the data source
+        Implement: Specific data source API calls, processing,
+        functionality required for retrieving data from the data source
 
         Args:
             stix_id (str): the id of the STIX 2.0 object to retrieve. Should
                 return a single object, the most recent version of the object
                 specified by the "id".
 
-            _composite_filters (list): list of filters passed along from
-                the Composite Data Filter.
+            _composite_filters (set): set of filters passed from the parent
```

Comment:
Well within the DataSource.get() call, only a STIX ID should be supplied. Thus any filters applied to the call are those attached to the DataSource or supplied from a CompositeDataSource.

If you pass in filters to a call, then that is a query (i.e. use query())

(or am I missing something?)

Reply:
Would it make sense to use `self.filters` then, instead of passing them as an argument to the function?

Reply:
I think the issue is for when the filters are defined on a "parent" CompositeDataSource, but also need to be applied to queries to child data sources.

The FileSystem and Memory data stores both just delegate calls to `get` to `query`. Maybe the right approach is to do something similar for TAXII (and make the case where the only filter is an `id=` a special case that calls `get_object` rather than `get_objects`), and then `get` only needs to be implemented on the base class as a proxy to `query`.

Reply:
Thats how TAXIICollectionSource.get() currently is. It just adds in any filters(attached to the Source or from CompositeDataSource)